### PR TITLE
Availability time picker

### PR DESCRIPTION
**Pull request by niusoff**

### CODE REVIEW

File: src/UXClient/Components/AvailabilityChart/AvailabilityChart.ts [hunk 1]
```diff
@@ -267,9 +267,8 @@ class AvailabilityChart extends ChartComponent{
 
             var dateTimeTextChildren = this.targetElement.select(".tsi-dateTimeContainer").selectAll("*");
             var pickerContainerChildren;
-            d3.select("body").on("click", () => {
+            this.targetElement.on("click", () => {
```
Comment (matdarMsft) [inline src/UXClient/Components/AvailabilityChart/AvailabilityChart.ts]: lets think about a named click handler here

File: src/UXClient/Components/AvailabilityChart/AvailabilityChart.ts [hunk 2]
```diff
@@ -451,14 +455,15 @@ class AvailabilityChart extends ChartComponent{
...
                 self.dateTimePickerContainer.style("display", "block");
-                var minMillis = self.fromMillis;
-                var maxMillis = self.toMillis
-                var startMillis = self.selectedFromMillis;
-                var endMillis = self.selectedToMillis;
-                self.dateTimePicker.render({'theme': self.chartOptions.theme}, minMillis, maxMillis, startMillis, endMillis, 
-                                        (fromMillis, toMillis) => {
-                                            self.dateTimePickerAction(fromMillis, toMillis);
-                                        });
+                var minMillis = self.fromMillis + (Utils.getOffsetMinutes(self.chartOptions.offset, self.fromMillis) * 60 * 1000);
```
Comment (matdarMsft) [inline src/UXClient/Components/AvailabilityChart/AvailabilityChart.ts]: lets test 00:50 to 2:05 in daylight savings just for giglez

File: src/UXClient/Models/ChartOptions.ts [hunk 3]
```diff
@@ -82,8 +83,8 @@ class ChartOptions {
         this.zeroYAxis = this.getValueOrDefault( chartOptionsObj, 'zeroYAxis', true);
         this.arcWidthRatio = this.getValueOrDefault(chartOptionsObj, 'arcWidthRatio', 0);
         this.brushClearable = this.getValueOrDefault(chartOptionsObj, 'brushClearable', true);
-        this.brushMoveAction = this.getValueOrDefault(chartOptionsObj, 'brushMoveAction', null);
-        this.brushMoveEndAction = this.getValueOrDefault(chartOptionsObj, 'brushMoveEndAction', null);
+        this.brushMoveAction = this.getValueOrDefault(chartOptionsObj, 'brushMoveAction', () => {});
```
Comment (matdarMsft) [inline src/UXClient/Models/ChartOptions.ts]: fix bug on how to determine if there is a brush or not

File: src/UXClient/Components/AvailabilityChart/AvailabilityChart.ts (hunk 1 above)
Comment (niusoff) [reply src/UXClient/Components/AvailabilityChart/AvailabilityChart.ts]: no Utils.createGUID :(

**Pull request merged**